docs/decisions/0001-custom-edx-proctoring-apis.rst

```diff
@@ -0,0 +1,35 @@
+1. Record Architecture Decisions
+--------------------------------
+
+Status
+------
+
+Review
+
+Context
+-------
+
+We would like to implement api that returns current sequence exam item and active exam attempt.
+Reasons:
+1. No api endpoint which can return submitted exam attempts by course_id, content_id or user_id.
+As single page application relies on attemt status provided by backend when user reloads the page
+after exam submission.
+2. It's more perfirmant and easier to use only one api endpoint to fetch all the needed data.
+3. We would like to get timed exam duration for the timer component so timer block is displayed and
+time is running in any course.
+
+Decision
+--------
+
+Create single api endpoint which provides requaired exam/attempt data:
```

Comment:
It seems to me `/edx_proctoring/v1/proctored_exam/attempt` it provides the same functionality as this new endpoint. Does it not?

example return
```
{
  "in_timed_exam": true,
  "taking_as_proctored": true,
  "exam_type": "an onboarding exam",
  "exam_display_name": "Exam",
  "exam_url_path": "/courses/course-v1:edX+UBW+3T2020/jump_to/block-v1:edX+UBW+3T2020+type@sequential+block@1e48c4a20624495f832c868dfbf90cd8",
  "time_remaining_seconds": 1784.660759,
  "low_threshold_sec": 360,
  "critically_low_threshold_sec": 90,
  "course_id": "course-v1:edX+UBW+3T2020",
  "attempt_id": 1,
  "accessibility_time_string": "you have 30 minutes remaining",
  "attempt_status": "started",
  "exam_started_poll_url": "/api/edx_proctoring/v1/proctored_exam/attempt/1",
  "desktop_application_js_url": "http://localhost:18000/static/bundles/%40edx/mockprock.js",
  "ping_interval": 60
}
```

Reply:
on further inspection I'm seeing the issue here I think, and that's because we want this endpoint to function when the user isn't actively taking the exam as well and that's why we need to pass course/content id.

A suggestion I might make is to try to keep these endpoints using the same existing logic and just have our new endpoint filter down by the course/content id. We could potentially even deprecate the old usage in the future to cut down on bloat.

Reply:
> because we want this endpoint to function when the user isn't actively taking the exam as well and that's why we need to pass course/content id. correct?

@Zacharis278 that's correct, i would say for both: for active taking and not active one



> A suggestion I might make is to try to keep these endpoints using the same existing logic and just have our new endpoint filter down by the course/content id.

We will investigate how and where to reuse repeating logic in APIs